src/test/java/seedu/linkedout/model/applicant/JobTest.java

```diff
@@ -0,0 +1,36 @@
+package seedu.linkedout.model.applicant;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.linkedout.testutil.Assert.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+public class JobTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new Job(null));
+    }
+
+    @Test
+    public void constructor_invalidJob_throwsIllegalArgumentException() {
+        String invalidJob = "";
+        assertThrows(IllegalArgumentException.class, () -> new Job(invalidJob));
+    }
+
+    @Test
+    public void isValidJob() {
+        // null job
+        assertThrows(NullPointerException.class, () -> Job.isValidJob(null));
+
+        // invalid jobs
+        assertFalse(Job.isValidJob("")); // empty string
+        assertFalse(Job.isValidJob(" ")); // spaces only
+
+        // valid jobs
+        assertTrue(Job.isValidJob("Software Engineer"));
+        assertTrue(Job.isValidJob("-")); // one character
```

Comment:
Do we want this case to pass? Might consider adding additional regex such that there is minimum one word in the Job Description